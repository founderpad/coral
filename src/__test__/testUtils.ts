// import { ThemeProvider } from "my-ui-lib"
// import { TranslationProvider } from "my-i18n-lib"
// import defaultStrings from "i18n/en-x-default"
import { render } from '@testing-library/react';
import { JSXElementConstructor, ReactElement } from 'react';

const Providers = ({ children }: { children: any }) => {
	return children;
	// return (
	//   <ThemeProvider theme="light">
	//     <TranslationProvider messages={defaultStrings}>
	//       {children}
	//     </TranslationProvider>
	//   </ThemeProvider>
	// )
};

const customRender = (
	ui: ReactElement<any, string | JSXElementConstructor<any>>,
	options = {}
): any => render(ui, { wrapper: Providers, ...options });

// re-export everything
export * from '@testing-library/react';
// override render method
export { customRender as render };
