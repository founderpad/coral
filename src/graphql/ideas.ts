import gql from 'graphql-tag';

const CREATE_IDEA = gql`
	mutation createIdea($idea: ideas_insert_input!) {
		idea: insert_ideas_one(object: $idea) {
			id
		}
	}
`;

const UPDATE_IDEA = gql`
	mutation updateIdea($idea: ideas_set_input!, $id: uuid!) {
		update_ideas_by_pk(pk_columns: { id: $id }, _set: $idea) {
			id
			name
			description
			field
			mission_statement
			competitors
			team
			additional_information
			is_published
			user_id
			status
		}
	}
`;

const DELETE_IDEA = gql`
	mutation deleteIdea($id: uuid!) {
		delete_ideas_by_pk(id: $id) {
			id
		}
	}
`;

const INSERT_INTERESTED_IDEA = gql`
	mutation insertInterestedIdea($ideaId: uuid!) {
		insert_interested_ideas_one(object: { idea_id: $ideaId }) {
			idea_id
		}
		update_ideas_by_pk(
			pk_columns: { id: $ideaId }
			_inc: { number_of_interested: 1 }
		) {
			id
		}
	}
`;

// const GET_IDEAS = gql`
// 	subscription getIdeas {
// 		ideas(
// 			order_by: { updated_at: desc }
// 			where: { is_published: { _eq: true } }
// 		) {
// 			id
// 			name
// 			description
// 			category
// 			user_id
// 			is_published
// 			created_at
// 		}
// 	}
// `;

const GET_IDEAS = gql`
	query getIdeas(
		$where: idea_preview_bool_exp
		$limit: Int
		$offset: Int
		$orderBy: [idea_preview_order_by!]
		$userId: uuid!
	) {
		idea_preview_aggregate(where: $where) {
			aggregate {
				count
			}
		}
		idea_preview(
			where: $where
			limit: $limit
			offset: $offset
			order_by: $orderBy
		) {
			id
			name
			preview
			field
			status
			createdAt
			isNew
			totalInterest
			totalComments
			idea_user {
				first_name
				country
				id
				avatar_url
				account {
					email
				}
			}
			idea_votes_aggregate {
				aggregate {
					count(columns: id)
				}
			}
			idea_votes(where: { user_id: { _eq: $userId } }) {
				id
			}
		}
	}
`;

const GET_USER_IDEAS = gql`
	query getUserIdeas($userId: uuid!) {
		user_ideas: ideas(
			where: { user_id: { _eq: $userId } }
			order_by: { created_at: desc }
		) {
			id
			name
			created_at
			is_published
			number_of_interested
			number_of_upvotes
			number_of_comments
		}
	}
`;

const GET_IDEA = gql`
	query getIdea($id: uuid!, $userId: uuid!) {
		idea: ideas_by_pk(id: $id) {
			id
			name
			description
			field
			competitors
			team
			additional_information
			is_published
			user_id
			status
			created_at
			number_of_interested
			number_of_comments
			number_of_upvotes
			idea_user {
				avatar_url
				first_name
				country
				id
			}
			idea_votes_aggregate {
				aggregate {
					count(columns: id)
				}
			}
		}
		has_interest: interested_ideas_by_pk(idea_id: $id, user_id: $userId) {
			id
		}
	}
`;

// const UPSERT_IDEA_UPVOTE = gql`
// 	mutation upsertIdeaVote($idea_vote: idea_votes_insert_input!) {
// 		insert_idea_votes_one(
// 			object: $idea_vote
// 			on_conflict: {
// 				constraint: idea_votes_pkey
// 				update_columns: vote_type
// 			}
// 		) {
// 			idea_id
// 		}
// 	}
// `;

const GET_IDEA_INTERESTED_USERS = gql`
	query getIdeaInterestedUsers($ideaId: uuid!) {
		interested_users: interested_ideas(
			where: { idea_id: { _eq: $ideaId } }
			order_by: { created_at: desc }
		) {
			id
			created_at
			user {
				display_name
				avatar_url
				account {
					email
				}
			}
		}
	}
`;

const INSERT_IDEA_UPVOTE = gql`
	mutation insertIdeaUpvote($idea_vote: idea_votes_insert_input!) {
		insert_idea_votes_one(object: $idea_vote) {
			id
			idea_id
		}
	}
`;

const DELETE_IDEA_UPVOTE = gql`
	mutation deleteIdeaUpvote($idea_id: uuid!, $user_id: uuid!) {
		delete_idea_votes(
			where: { user_id: { _eq: $user_id }, idea_id: { _eq: $idea_id } }
		) {
			affected_rows
		}
	}
`;

export {
	CREATE_IDEA,
	GET_USER_IDEAS,
	GET_IDEAS,
	GET_IDEA,
	UPDATE_IDEA,
	DELETE_IDEA,
	// UPSERT_IDEA_UPVOTE,
	INSERT_INTERESTED_IDEA,
	GET_IDEA_INTERESTED_USERS,
	INSERT_IDEA_UPVOTE,
	DELETE_IDEA_UPVOTE
};
