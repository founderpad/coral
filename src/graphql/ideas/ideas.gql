#import { UserFieldsWithEmail } from '../fragments/user.gql'

mutation CreateIdea($idea: ideas_insert_input!) {
	idea: insert_ideas_one(object: $idea) {
		id
		name
	}
}

mutation UpdateIdea($idea: ideas_set_input!, $id: uuid!) {
	update_ideas_by_pk(pk_columns: { id: $id }, _set: $idea) {
		id
		name
		summary
		description
		field
		competitors
		team
		additionalInformation
		isPublished
		userId
		status
	}
}

mutation DeleteIdea($id: uuid!) {
	delete_ideas_by_pk(id: $id) {
		id
	}
}

mutation CreateInterestedIdea($ideaId: uuid!, $targetUserId: uuid!) {
	addInterestedIdea: insert_interested_ideas_one(
		object: { ideaId: $ideaId, targetUserId: $targetUserId }
	) {
		ideaId
	}
	# updateIdeaTotalInterested: update_ideas_by_pk(
	# 	pk_columns: { id: $ideaId }
	# 	_inc: { totalInterested: 1 }
	# ) {
	# 	id
	# }
}

query Ideas(
	$where: idea_preview_bool_exp
	$limit: Int
	$offset: Int
	$orderBy: [idea_preview_order_by!]
	$userId: uuid!
) {
	idea_preview_aggregate(where: $where) {
		aggregate {
			count
		}
	}
	idea_preview(
		where: $where
		limit: $limit
		offset: $offset
		order_by: $orderBy
	) {
		...IdeaPreviewFields
	}
}

fragment BoostedIdeaFields on boosted_ideas {
	ideaId
	totalCurrencyAmount
	remainingCurrencyAmount
	createdAt
	idea {
		name
		summary
		status
		field
	}
}

fragment IdeaPreviewFields on idea_preview {
	id
	name
	field
	status
	createdAt
	isNew
	isPublished
	summary
	userId
	user {
		...UserFieldsWithEmail
	}
	votes_aggregate {
		aggregate {
			count(columns: id)
		}
	}
	comments_aggregate {
		aggregate {
			count(columns: id)
		}
	}
	interested_aggregate {
		aggregate {
			count(columns: id)
		}
	}
	votes(where: { userId: { _eq: $userId } }) {
		id
	}
}

query UserIdeas($userId: uuid!) {
	# user_ideas: ideas(
	# 	where: { userId: { _eq: $userId } }
	# 	order_by: { createdAt: desc }
	# ) {
	# 	id
	# 	name
	# 	createdAt
	# 	isPublished
	# 	totalInterested
	# 	totalVotes
	# 	totalComments
	# }

	user_ideas: idea_preview(
		where: { userId: { _eq: $userId } }
		order_by: { createdAt: desc }
	) {
		...IdeaPreviewFields
	}
}

fragment IdeaFields on ideas {
	id
	summary
	name
	description
	field
	competitors
	team
	additionalInformation
	isPublished
	userId
	status
	createdAt
	totalInterested
	totalComments
	totalVotes
}

query Idea($id: uuid!, $userId: uuid!) {
	idea: ideas_by_pk(id: $id) {
		...IdeaFields
		user {
			...UserFieldsWithEmail
		}
		votes_aggregate {
			aggregate {
				count(columns: id)
			}
		}
		votes(where: { userId: { _eq: $userId } }) {
			id
		}
		interested_aggregate {
			aggregate {
				count(columns: id)
			}
		}
	}
	hasInterest: interested_ideas_by_pk(ideaId: $id, userId: $userId) {
		...InterestedIdeaFields
	}
}

fragment InterestedIdeaFields on interested_ideas {
	id
	ideaId
	userId
}

# fragment InterestedIdea on interested_ideas {
# 	hasInterest: interested_ideas_by_pk(ideaId: $id, userId: $userId) {
# 		id
# 	}
# }

query IdeaInterestedUsers($ideaId: uuid!) {
	interested_users: interested_ideas(
		where: { ideaId: { _eq: $ideaId } }
		order_by: { createdAt: desc }
	) {
		id
		createdAt
		user {
			...UserFieldsWithEmail
		}
	}
}

mutation InsertIdeaUpvote($idea_vote: idea_votes_insert_input!) {
	insert_idea_votes_one(object: $idea_vote) {
		id
		ideaId
	}
}

mutation DeleteIdeaUpvote($ideaId: uuid!, $userId: uuid!) {
	delete_idea_votes(
		where: { userId: { _eq: $userId }, ideaId: { _eq: $ideaId } }
	) {
		affected_rows
	}
}

query BoostedIdeas($limit: Int, $offset: Int) {
	boosted_ideas(limit: 10, offset: 0) {
		...BoostedIdeaFields
	}
}
