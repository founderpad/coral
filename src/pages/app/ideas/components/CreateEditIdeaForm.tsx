import { Stack } from '@chakra-ui/layout';
import {
	Alert,
	AlertDescription,
	AlertIcon,
	AlertTitle,
	Box,
	CloseButton,
	Link
} from '@chakra-ui/react';
import { SubmitButton } from 'components/buttons';
import { Form } from 'components/form';
import { InputField } from 'components/input/InputField';
import { SelectField } from 'components/input/SelectField';
import { SwitchField } from 'components/input/SwitchField';
import { TextareaField } from 'components/input/TextareaField';
import { TIdeas, useCreateIdeaMutation } from 'generated/api';
import NextLink from 'next/link';
import React, { useEffect } from 'react';
import { useForm } from 'react-hook-form';
import { ideasStatusList, industriesList } from 'utils/Constants';

type TEditIdea = Omit<TIdeas, 'idea_user' | 'idea_votes'>;

const CreateEditIdeaForm = ({ idea }: { idea?: TIdeas }): JSX.Element => {
	const {
		handleSubmit,
		control,
		getValues,
		reset,
		formState: { errors, isSubmitting, isValid }
	} = useForm<TEditIdea>({
		mode: 'all',
		defaultValues: {
			...idea,
			is_published: true
		}
	});

	const [createIdeaMutation, { data: createSucess }] = useCreateIdeaMutation({
		variables: {
			idea: getValues() // value for 'idea'
		}
	});

	useEffect(() => {
		if (createSucess) {
			reset({
				name: '',
				mission_statement: '',
				description: '',
				field: '',
				competitors: '',
				team: '',
				additional_information: '',
				status: ''
			});
		}
	}, [createSucess]);

	return (
		<Form
			id={'create-edit-idea-form'}
			name={'create-edit-Create edit idea form-form'}
			onSubmit={handleSubmit(createIdeaMutation)}
			noValidate
		>
			{' '}
			<Stack spacing={8} d={'flex'} bg={'white'}>
				<InputField
					id="name"
					name="name"
					label="Name of your idea"
					placeholder="Name of your idea"
					error={errors['name']}
					errorText="Please enter a name for your idea"
					control={control}
					isRequired
				/>

				<TextareaField
					id="description"
					name="description"
					label="Description of your idea"
					placeholder="Write a description about your idea (max. 500 characters)"
					error={errors['description']}
					errorText="Please enter your idea description (max. 500 characters)"
					control={control}
					rules={{ maxLength: 500 }}
					isRequired
				/>

				{/* <TextareaField
					id="missionstatement"
					label="Mission statement"
					name="mission_statement"
					placeholder="Write a mission statement about your idea (max. 500 characters)"
					control={control}
					rules={{ maxLength: 500 }}
				/> */}

				<SelectField
					id="field"
					name="field"
					label="What field is your idea?"
					error={errors['field']}
					errorText="Please select the field for your idea."
					placeholder="field"
					size={'md'}
					options={industriesList()}
					control={control}
					isRequired
				/>

				<SelectField
					id="status"
					name="status"
					label="What is its current status?"
					error={errors['status']}
					errorText="Please select the status for your idea."
					placeholder="status"
					size={'md'}
					options={ideasStatusList()}
					control={control}
					isRequired
				/>

				<TextareaField
					id="competitors"
					name="competitors"
					label="Your competitors"
					placeholder="List your competitors about your idea"
					control={control}
					// rules={{ maxLength: 150 }}
				/>

				<TextareaField
					id="team"
					label="Your team"
					name="team"
					placeholder="List each team member"
					control={control}
					// rules={{ maxLength: 150 }}
				/>

				<TextareaField
					id="additional_information"
					label="Additional information"
					name="additional_information"
					placeholder="Any additional information"
					control={control}
					// rules={{ maxLength: 150 }}
				/>

				{/* <FormControl>
					<Heading as={'h5'} size={'sm'} color={'fpGrey.900'}>
						Supporting material
					</Heading>
					<Text fontSize={'sm'} color={'fpGrey.500'}>
						Any supporting material you wish to share such as pitch
						decks, business plans and accelerators.
					</Text>
					<FormHelperText>
						<em>Max. file size: 5mb</em>
						<br />
						<em>Supported types are .doc, .docx, .pdf, .txt</em>
					</FormHelperText>
				</FormControl>

				<Stack
					w={'full'}
					flex={1}
					justifyContent={'flex-start'}
					spacing={4}
					direction={{ base: 'column', sm: 'row' }}
					pb={4}
				>
					<FileUploader label="Business Plan" />
					<FileUploader label="Pitch Deck" />
					<FileUploader label="Other" />
				</Stack> */}

				<SwitchField
					id="is_published"
					name="is_published"
					label="Publish your idea"
					helperText="You can change this after it's been created"
					defaultChecked={true}
					control={control}
				/>

				{/* {(updateSuccess || createSucess) && (
					<CreateUpdateSuccessAlert
						isUpdate={!!updateSuccess}
						ideaId={updateSuccess ? idea.id : createSucess.idea.id}
					/>
				)} */}

				{createSucess && (
					<CreateUpdateSuccessAlert
						isUpdate={false}
						ideaId={createSucess.idea.id}
					/>
				)}

				<SubmitButton
					name={'create-idea-button'}
					label={(idea ? 'Update your ' : 'Create your ') + 'idea'}
					alignSelf={'center'}
					isLoading={isSubmitting}
					disabled={!isValid || isSubmitting}
					mt={'auto'}
					minW={'xs'}
					size={'md'}
				/>
			</Stack>
		</Form>
	);
};

const CreateUpdateSuccessAlert = ({
	isUpdate,
	ideaId
}: {
	isUpdate: boolean;
	ideaId: string;
}): JSX.Element => {
	return (
		<Alert status="success" variant={'left-accent'}>
			<AlertIcon />
			<Box flex="1">
				<AlertTitle color={'fpGrey.900'} fontWeight={'bold'}>
					Your idea has been {isUpdate ? 'updated' : 'created'}
				</AlertTitle>
				<AlertDescription display="block" fontSize={'sm'} mb={4}>
					Your {isUpdate ? 'updated' : 'new'} idea has been added and
					will be live to other founders.
				</AlertDescription>

				<NextLink href={`/app/idea/${ideaId}`} passHref>
					<Link color={'fpPrimary.500'} fontSize={'sm'}>
						Click here to view your idea
					</Link>
				</NextLink>
			</Box>
			<CloseButton position="absolute" right="8px" top="8px" />
		</Alert>
	);
};

export default CreateEditIdeaForm;
