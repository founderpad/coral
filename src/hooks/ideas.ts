import {
    MutationTuple,
    QueryResult,
    useMutation,
    useQuery
} from '@apollo/client';
import { TIdeas } from 'generated/graphql';
import {
    CREATE_IDEA,
    DELETE_IDEA,
    GET_IDEA,
    GET_IDEAS,
    GET_USER_IDEAS
} from 'graphql/ideas';
import { useRouter } from 'next/router';
import { TIdea, TIdeaPreview } from 'types/idea';
import { useQueryParam } from './util';

export const useCreateIdea = (idea: TIdea): MutationTuple<any, any> => {
	return useMutation(CREATE_IDEA, {
		variables: { idea }
		// update: updateCache
	});
};

export const useGetUserIdeas = (
	userId: string
): QueryResult<TIdeaPreview[], any> => {
	const response = useQuery(GET_USER_IDEAS, {
		variables: { user_id: userId }
	});

	return { ...response, data: response.data?.ideas as TIdeaPreview[] };
};

export const useGetIdeas = (): QueryResult<{
	ideas: TIdeaPreview[];
	total: number;
}> => {
	const router = useRouter();
	const { name: searchTerm, field: searchIndustry, is_new } = router.query;
	const page = parseInt(useQueryParam('page'));

	const response = useQuery(GET_IDEAS, {
		variables: {
			offset: (page - 1) * 3,
			name: searchTerm ? '%' + searchTerm + '%' : undefined,
			field: searchIndustry ? searchIndustry : undefined,
			is_new: is_new ?? undefined
		},
		fetchPolicy: 'no-cache'
	});

	return {
		...response,
		data: {
			ideas: response.data?.idea_preview,
			total: response.data?.idea_preview_aggregate.aggregate.count
		}
	};
};

export const useGetIdea = (id: string): QueryResult<TIdeas, any> => {
	const response = useQuery(GET_IDEA, {
		variables: { id }
	});

	return { ...response, data: response.data?.idea };
};

export const useDeleteIdea = (id: string): any => {
	return useMutation(DELETE_IDEA, {
		variables: { id }
	});
};
