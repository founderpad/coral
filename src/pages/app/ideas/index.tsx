import { Grid, GridItem } from '@chakra-ui/react';
import { PageLayout, StackLayout } from 'components/layouts';
import { DocumentTitle } from 'components/shared';
import React from 'react';
import AuthFilter from 'utils/AuthFilter';
import IdeasSearchForm from './components/IdeasSearchForm';
import IdeasContainer from './IdeasContainer';

const TIdeas = (): JSX.Element => (
	<React.Fragment>
		<DocumentTitle title="All ideas" />
		<PageLayout
			title="All ideas"
			subtitle="The latest ideas from other users"
		>
			<Grid templateColumns="repeat(7, 1fr)" gap={6} minH={'full'}>
				<GridItem
					display={{ base: 'none', md: 'block' }}
					colSpan={{ md: 3, lg: 2 }}
					position={'relative'}
				>
					<DesktopSearch />
				</GridItem>

				<GridItem
					colSpan={{ base: 7, md: 4, lg: 5 }}
					position={'relative'}
				>
					<IdeasContainer />
				</GridItem>
			</Grid>
		</PageLayout>
	</React.Fragment>
);

const DesktopSearch = () => (
	<StackLayout
		display={{ base: 'none', md: 'flex' }}
		pr={6}
		h={'full'}
		bg={'white'}
		spacing={1}
		borderRightWidth={1}
		borderRightColor={'gray.100'}
		rounded={'none'}
	>
		<IdeasSearchForm />
	</StackLayout>
);

export default AuthFilter(TIdeas);
