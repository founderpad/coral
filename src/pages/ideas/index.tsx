import { Grid, GridItem } from '@chakra-ui/react';
import { PageLayout, StackLayout } from 'components/layouts';
import WindowTitle from 'components/shared/WindowTitle';
import React from 'react';
import AuthFilter from 'utils/AuthFilter';
import IdeasSearchForm from './components/IdeasSearchForm';
import IdeasContainer from './IdeasContainer';

const Ideas = (): JSX.Element => (
	<React.Fragment>
		<WindowTitle title="All ideas" />
		<PageLayout
			title="All ideas"
			subtitle="The latest ideas from other users"
		>
			<Grid templateColumns="repeat(7, 1fr)" gap={6} minH={'full'}>
				<GridItem
					display={{ base: 'none', md: 'block' }}
					colSpan={{ md: 3, lg: 2 }}
					position={'relative'}
				>
					<DesktopSearch />
				</GridItem>

				<GridItem
					colSpan={{ base: 7, md: 4, lg: 5 }}
					position={'relative'}
				>
					<IdeasContainer />
				</GridItem>
			</Grid>
		</PageLayout>
	</React.Fragment>
);

const DesktopSearch = () => (
	<StackLayout
		display={{ base: 'none', md: 'flex' }}
		pr={6}
		h={'full'}
		bg={'white'}
		borderRightWidth={1}
		borderRightColor={'fpLightGrey.300'}
	>
		<IdeasSearchForm />
	</StackLayout>
);

export default AuthFilter(Ideas);
