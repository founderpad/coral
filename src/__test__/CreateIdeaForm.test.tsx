import { MockedProvider } from '@apollo/client/testing';
import { fireEvent } from '@testing-library/dom';
import { TIdeas } from 'generated/graphql';
import { CREATE_IDEA } from 'graphql/ideas';
import CreateEditIdeaForm from 'pages/ideas/components/CreateEditIdeaForm';
import { render } from '__test__/testUtils';

describe('Create idea form', () => {
	const idea: Partial<TIdeas> = {
		name: 'Idea name',
		description: 'Some idea description',
		additional_information: 'Some additional information',
		business_plan: 'Some business plan',
		team: 'Some team',
		competitors: 'Some competitors',
		mission_statement: 'Some mission statement',
		field: 'Finance',
		is_published: true
	};

	const mocks = [
		{
			request: {
				query: CREATE_IDEA,
				variables: {
					idea
				}
			},
			result: {
				data: {
					idea: {
						id: '2',
						name: 'Idea name',
						description: 'Some idea description',
						additional_information: 'Some additional information',
						business_plan: 'Some business plan',
						team: 'Some team',
						competitors: 'Some competitors',
						mission_statement: 'Some mission statement',
						field: 'Finance',
						user_id: '1',
						is_published: true
					}
				}
			}
		}
	];

	it('matches snapshot', () => {
		const { asFragment } = render(
			<MockedProvider mocks={mocks} addTypename={false}>
				<CreateEditIdeaForm />
			</MockedProvider>
		);
		expect(asFragment()).toMatchSnapshot();
	});

	it('renders CreateEditIdea form', () => {
		const { getByText } = render(
			<MockedProvider mocks={mocks} addTypename={false}>
				<CreateEditIdeaForm />
			</MockedProvider>
		);
		expect(getByText(/Publish your idea/i)).toBeInTheDocument();
	});

	it('should render CreateEditIdeaForm fields', () => {
		const { getByRole } = render(
			<MockedProvider mocks={mocks} addTypename={false}>
				<CreateEditIdeaForm />
			</MockedProvider>
		);

		expect(getByRole('textbox', { name: /name/i })).toBeInTheDocument();
		expect(
			getByRole('textbox', { name: /description/i })
		).toBeInTheDocument();
		// expect(
		// 	getByRole('textbox', { name: /mission_statement/i })
		// ).toBeInTheDocument();
		// expect(getByRole('combobox', { name: /field/i })).toBeInTheDocument();
		expect(getByRole('combobox', { name: /status/i })).toBeInTheDocument();
		expect(
			getByRole('textbox', { name: /competitors/i })
		).toBeInTheDocument();
		expect(getByRole('textbox', { name: /team/i })).toBeInTheDocument();
		// expect(getByRole('textbox', { name: /additional_information/i })).toBeInTheDocument();
		// expect(getByRole('switch', { name: /is_published/i })).toBeInTheDocument();
	});

	it('should validate CreateEditIdeaForm fields', () => {
		const mockSaveIdea = jest.fn();
		const { getByRole } = render(
			<MockedProvider mocks={mocks} addTypename={false}>
				<CreateEditIdeaForm />
			</MockedProvider>
		);

		fireEvent.input(getByRole('textbox', { name: /name/i })),
			{
				target: {
					value: 'A new idea name'
				}
			};

		fireEvent.input(getByRole('textbox', { name: /description/i })),
			{
				target: {
					value: 'A new idea description'
				}
			};

		fireEvent.submit(getByRole('button'));
		expect(mockSaveIdea).not.toBeCalled();
	});

	// it('should populate CreateEditIdeaForm fields', () => {
	// 	const { getByRole } = render(
	// 		<MockedProvider mocks={mocks} addTypename={false}>
	// 			<CreateEditIdeaForm idea={idea} />
	// 		</MockedProvider>
	// 	);

	// 	expect(getByRole('form')).toHaveFormValues({
	// 		name: 'Idea name',
	// 		description: 'Some idea description',
	// 		additional_information: 'Some additional information',
	// 		business_plan: 'Some business plan',
	// 		team: 'Some team',
	// 		competitors: 'Some competitors',
	// 		mission_statement: 'Some mission statement',
	// 		field: 'Finance',
	// 		is_published: true
	// 	});
	// })
});
