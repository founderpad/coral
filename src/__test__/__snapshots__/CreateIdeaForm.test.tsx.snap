// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create idea form matches snapshot 1`] = `
<DocumentFragment>
  <form
    aria-label="create-edit-Create edit idea form-form"
    autocomplete="off"
    id="create-edit-idea-form"
    name="create-edit-Create edit idea form-form"
    novalidate=""
  >
    <div
      class="chakra-stack css-km7s47"
    >
      <div
        class="chakra-stack css-1uxng5d"
      >
        <div
          class="chakra-form-control css-1naxzcr"
          role="group"
        >
          <label
            class="chakra-form__label css-1kr1wuy"
            for="name"
            id="name-label"
          >
            Name of your idea
            <span
              aria-hidden="true"
              class="chakra-form__required-indicator css-0"
              role="presentation"
            >
              *
            </span>
          </label>
          <input
            aria-label="name"
            aria-required="true"
            class="chakra-input css-0"
            control="[object Object]"
            errortext="Please enter a name for your idea"
            id="name"
            label="Name of your idea"
            name="name"
            placeholder="Name of your idea"
            required=""
            value=""
          />
        </div>
        <div
          class="chakra-form-control css-0"
          role="group"
        >
          <label
            class="chakra-form__label css-1kr1wuy"
            for="description"
            id="description-label"
          >
            Description of your idea
            <span
              aria-hidden="true"
              class="chakra-form__required-indicator css-0"
              role="presentation"
            >
              *
            </span>
          </label>
          <textarea
            aria-required="true"
            class="chakra-textarea css-1xh66r"
            id="description"
            placeholder="Write a description about your idea (max. 500 characters)"
            required=""
          />
        </div>
        <div
          class="chakra-form-control css-m5v8bu"
          role="group"
        >
          <label
            class="chakra-form__label css-1kr1wuy"
            for="field"
            id="field-label"
          >
            What field is your idea?
            <span
              aria-hidden="true"
              class="chakra-form__required-indicator css-0"
              role="presentation"
            >
              *
            </span>
          </label>
          <div
            class="chakra-select__wrapper css-42b2qy"
          >
            <select
              aria-label="select-field-field"
              aria-required="true"
              class="chakra-select css-1j1jvqa"
              id="select-field-field"
              name="select-field-field"
              required=""
            >
              <option
                value=""
              >
                Select field
              </option>
              <option
                value="Agriculture"
              >
                Agriculture
              </option>
              <option
                value="Education"
              >
                Education
              </option>
              <option
                value="Engineering"
              >
                Engineering
              </option>
              <option
                value="Food & Drink"
              >
                Food & Drink
              </option>
              <option
                value="Media"
              >
                Media
              </option>
              <option
                value="Medical"
              >
                Medical
              </option>
              <option
                value="Online Services"
              >
                Online Services
              </option>
              <option
                value="Other"
              >
                Other
              </option>
              <option
                value="Professional Services"
              >
                Professional Services
              </option>
              <option
                value="Retail"
              >
                Retail
              </option>
              <option
                value="Sports"
              >
                Sports
              </option>
              <option
                value="Technology"
              >
                Technology
              </option>
              <option
                value="Tourism"
              >
                Tourism
              </option>
              <option
                value="Transport"
              >
                Transport
              </option>
              <option
                value="Utilities"
              >
                Utilities
              </option>
            </select>
            <div
              class="chakra-select__icon-wrapper css-162mkon"
            >
              <svg
                aria-hidden="true"
                class="chakra-select__icon"
                focusable="false"
                role="presentation"
                style="width: 1em; height: 1em; color: currentColor;"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                  fill="currentColor"
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          class="chakra-form-control css-m5v8bu"
          role="group"
        >
          <label
            class="chakra-form__label css-1kr1wuy"
            for="status"
            id="status-label"
          >
            What is its current status?
            <span
              aria-hidden="true"
              class="chakra-form__required-indicator css-0"
              role="presentation"
            >
              *
            </span>
          </label>
          <div
            class="chakra-select__wrapper css-42b2qy"
          >
            <select
              aria-label="select-status-field"
              aria-required="true"
              class="chakra-select css-1j1jvqa"
              id="select-status-field"
              name="select-status-field"
              required=""
            >
              <option
                value=""
              >
                Select status
              </option>
              <option
                value="New idea"
              >
                New idea
              </option>
              <option
                value="Building team"
              >
                Building team
              </option>
              <option
                value="Pre-launch"
              >
                Pre-launch
              </option>
              <option
                value="Launched"
              >
                Launched
              </option>
            </select>
            <div
              class="chakra-select__icon-wrapper css-162mkon"
            >
              <svg
                aria-hidden="true"
                class="chakra-select__icon"
                focusable="false"
                role="presentation"
                style="width: 1em; height: 1em; color: currentColor;"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                  fill="currentColor"
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          class="chakra-form-control css-0"
          role="group"
        >
          <label
            class="chakra-form__label css-1kr1wuy"
            for="competitors"
            id="competitors-label"
          >
            Your competitors
          </label>
          <textarea
            class="chakra-textarea css-1xh66r"
            id="competitors"
            placeholder="List your competitors about your idea"
          />
        </div>
        <div
          class="chakra-form-control css-0"
          role="group"
        >
          <label
            class="chakra-form__label css-1kr1wuy"
            for="team"
            id="team-label"
          >
            Your team
          </label>
          <textarea
            class="chakra-textarea css-1xh66r"
            id="team"
            placeholder="List each team member"
          />
        </div>
        <div
          class="chakra-form-control css-0"
          role="group"
        >
          <label
            class="chakra-form__label css-1kr1wuy"
            for="additional_information"
            id="additional_information-label"
          >
            Additional information
          </label>
          <textarea
            class="chakra-textarea css-1xh66r"
            id="additional_information"
            placeholder="Any additional information"
          />
        </div>
        <div
          class="chakra-form-control css-0"
          role="group"
        >
          <label
            class="chakra-form__label css-as19zo"
            for="is_published"
            id="label-is_published"
          >
            Publish your idea
          </label>
          <label
            class="chakra-switch css-evm8da"
            data-checked=""
            role="switch"
          >
            <input
              checked=""
              class="chakra-switch__input"
              id="is_published"
              style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
              type="checkbox"
              value="true"
            />
            <span
              aria-hidden="true"
              class="chakra-switch__track css-14qxnv8"
              data-checked=""
            >
              <span
                class="chakra-switch__thumb css-0"
                data-checked=""
              />
            </span>
          </label>
          <div
            class="chakra-form__helper-text css-0"
            id="helper-text-is_published"
          >
            You can change this after it's been created
          </div>
        </div>
        <button
          aria-label="create-idea-button"
          class="chakra-button css-i8vml2"
          disabled=""
          type="submit"
        >
          Create your idea
        </button>
      </div>
    </div>
  </form>
</DocumentFragment>
`;
